code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -351,9 +351,10 @@
 
 
 /mob/living/simple_animal/hostile/proc/AttackingTarget(atom/attacked_target)
-	SEND_SIGNAL(src, COMSIG_HOSTILE_ATTACKINGTARGET, target)
 	in_melee = TRUE
-	return target.attack_animal(src)
+	var/result = target.attack_animal(src)
+	SEND_SIGNAL(src, COMSIG_HOSTILE_ATTACKINGTARGET, target, result)
```

Comment:
It's better if this is two separate signals--one for trying the attack (what it was before), and one for doing the attack (here).

This removes a needless branch parameter, but more importantly lets us easily implement cancelling as well.